packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -245,7 +246,17 @@ class AndroidStudio implements Comparable<AndroidStudio> {
     }
     AndroidStudio? newest;
     for (final AndroidStudio studio in studios.where((AndroidStudio s) => s.isValid)) {
-      if (newest == null || studio.compareTo(newest) > 0) {
+      newest ??= studio;
+
+      if (studio.version != null && newest.version == null) {
+        newest = studio;
+      }
+      else if (studio.version == null && newest.version == null &&
+            studio.directory.compareTo(newest.directory) > 0 ) {
```

Comment:
nit. I think you can directly compare these directories with `>`

Reply:
These are instances of `String`, which don't implement the `>` operator.